thesis-gui/src/components/private-quiz/PrivateQuiz.tsx

```diff
@@ -0,0 +1,178 @@
+import React from "react";
+import { useEffect, useState } from "react";
+import Answer from "../answer/Answer";
+import {quizService} from "../../services/quiz-service";
+import {
+    IonButton, IonCard,
+    IonCardContent,
+    IonCardHeader,
+    IonCardSubtitle,
+    IonCardTitle,
+    IonContent,
+    IonList,
+    IonLoading,
+    IonPage,
+    IonRadioGroup, IonRouterLink,
+} from "@ionic/react";
+import {PageHeader} from "../common/PageHeader";
+import {QuestionsSetResponse, TeacherQuestionResponse} from "../../types/my-quiz";
+import {useParams} from "react-router";
+
+const PAGE_NAME = "Private Quiz";
+const NEW_QUESTION_DELAY_MS = 1500;
+
+const WrongKeyErrorCard: React.FC<{ secretKey: string }> = ({ secretKey }) => {
+    return (
+        <IonCard>
+            <IonCardHeader>
+                <IonCardSubtitle>Error</IonCardSubtitle>
+                <IonCardTitle>Wrong Key</IonCardTitle>
+            </IonCardHeader>
+
+            <IonCardContent>
+                { `There's no private quiz with secret key '${secretKey}'!\nGet the right one and try again ` }
+                <IonRouterLink routerLink="/private-quiz">here</IonRouterLink>
+            </IonCardContent>
+        </IonCard>
+    );
+}
+
+const QuizResultCard: React.FC<{
+    questionsSetName: string,
+    correctAnswers: number,
+    numberOfQuestions: number
+}> = ({
+        questionsSetName,
+        correctAnswers,
+        numberOfQuestions
+}) => {
+    return (
+        <IonCard>
+            <IonCardHeader>
+                <IonCardSubtitle>
+                    { `Quiz: ${ questionsSetName }` }
+                </IonCardSubtitle>
+                <IonCardTitle>
+                    You finished the quiz!
+                </IonCardTitle>
+            </IonCardHeader>
+            <IonCardContent>
+                { `Your result is: ${correctAnswers} / ${numberOfQuestions} (${Math.round(correctAnswers / numberOfQuestions * 100)}%)` }
```

Comment:
I would extract the result of this computation to a variable with a good name.

Reply:
Done